cmd/proxy/actions/app.go

```diff
@@ -72,10 +72,11 @@ func App(conf *config.Config) (http.Handler, error) {
 		// can not send custom headers and therefore if the proxy
 		// is running behind a prefix as well as some authentication
 		// mechanism, we should allow the plain / to return 200.
-		r.HandleFunc("/", healthHandler).Methods(http.MethodGet)
 		subRouter = r.PathPrefix(prefix).Subrouter()
 	}
 
+	r.PathPrefix("/healthz").Subrouter()
```

Comment:
@manugupt1 you're not adding any routes to this subrouter, so I'm not sure this will add any additional routes under `/healthz`

What do you think about adding the ability to skip basic auth checks for specific routes in `basicAuth` middleware, and using that to skip basic auth on `/healthz`?